tools/benchmark.sh

```diff
@@ -164,37 +232,55 @@ function summarize_result {
   # that "Compaction Stats" is written to stdout at least once. If it won't
   # happen then empty output from grep when searching for "Sum" will cause
   # syntax errors.
+  version=$( grep ^RocksDB: $test_out | awk '{ print $3 }' )
+  date=$( grep ^Date: $test_out | awk '{ print $6 "-" $3 "-" $4 "T" $5 ".000" }' )
+  iso_date=$( month_to_num $date )
+  tz=$( date "+%z" )
+  iso_tz="${tz:0:3}:${tz:3:2}"
+  iso_date="$iso_date$iso_tz"
   uptime=$( grep ^Uptime\(secs $test_out | tail -1 | awk '{ printf "%.0f", $2 }' )
   stall_time=$( grep "^Cumulative stall" $test_out | tail -1  | awk '{  print $3 }' )
   stall_pct=$( grep "^Cumulative stall" $test_out| tail -1  | awk '{  print $5 }' )
   ops_sec=$( grep ^${bench_name} $test_out | awk '{ print $5 }' )
   mb_sec=$( grep ^${bench_name} $test_out | awk '{ print $7 }' )
-  lo_wgb=$( grep "^  L0" $test_out | tail -1 | awk '{ print $9 }' )
+  l0_wgb=$( grep "^  L0" $test_out | tail -1 | awk '{ print $9 }' )
   sum_wgb=$( grep "^ Sum" $test_out | tail -1 | awk '{ print $9 }' )
   sum_size=$( grep "^ Sum" $test_out | tail -1 | awk '{ printf "%.1f", $3 / 1024.0 }' )
-  if [ "$lo_wgb" = "" ]; then
-      lo_wgb="0.0"
+  if [[ "$l0_wgb" == "" ]]; then
+      l0_wgb="0.0"
   fi
-  if [ "$lo_wgb" = "0.0" ]; then
+  if [[ "$l0_wgb" == "0.0" ]]; then
       wamp="0.0"
   else
-      wamp=$( echo "scale=1; $sum_wgb / $lo_wgb" | bc )
+      wamp=$( echo "scale=1; $sum_wgb / $l0_wgb" | bc )
+  fi
+  if [[ "$sum_wgp" == "" ]]; then
+      wmb_ps=""
```

Comment:
It was actually already previously `""` when `$sum_wgp == ""` becuase the following line previously raised an error:
```
wmb_ps=$( echo "scale=1; ( $sum_wgb * 1024.0 ) / $uptime" | bc )
```

All I have done is to stop the error occurring and its error message being shown in the output from `benchmark.sh` ;-)

Reply:
It would still be helpful to set it to "0" or "??" or something.  Otherwise any awk-like script will have trouble parsing the output.

Reply:
@mrambacher As the output is TSV, it is quite possible to parse it with awk. I think it is important to differentiate between null (i.e. no data available) and the value 0 or the empty string. They mean different things...

When processing a TSV with awk, you can use something like the following to correctly output the value of each column, even those which have no value (e.g. null).

```
tail -1  /tmp/report.tsv | awk -F'\t' 'BEGIN {OFS = FS} {print $8}'
```

I think it might actually be good to go further and make use of quotes around strings, in this way it becomes very clear on how to differentiate between: (a) a non-empty string, (b) an empty string, and (c) no value (i.e. null).

Hopefully the above allays your concerns about awk?